### PR TITLE
chore: use EDC M5 version

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

Uses the latest EDC M5 version in local and CI builds. Removes the local checkout entirely.
ALso fixes a broken test.

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/identityservice/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/identityservice/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -3,7 +3,7 @@ projectGroup=org.eclipse.dataspaceconnector.identityhub
 defaultVersion=0.0.1-SNAPSHOT
 
 edcGroup=org.eclipse.dataspaceconnector
-edcVersion=0.0.1-SNAPSHOT
+edcVersion=0.0.1-milestone-5
```
Comment (cpeeyush) [inline gradle.properties:6]: @paullatzelsperger is this change needed, Because i see even on milestone-5 tag EDC package version is still 0.0.1-SNAPSHOT
 
https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/39e8da36860a9087eae76fcd3cdae2023b4e30d0/gradle.properties#L2

or maybe I am missing something, good to see your though on this. Thanks.

Comment (paullatzelsperger) [reply gradle.properties:6]: the package version of EDC gets overridden during the release process, so yes, it is needed.

**Comment by marcgs** [comment]
Suggesting an alternative fix for the broken test: https://github.com/agera-edc/IdentityHub/pull/45

File: gradle.properties (hunk 1 above)
Comment (paullatzelsperger) [reply gradle.properties:6]: edit: updated description to also inform about the removing of the local checkout/build

**Comment by paullatzelsperger** [comment]
> Suggesting an alternative fix for the broken test: [agera-edc#45](https://github.com/agera-edc/IdentityHub/pull/45)

@marcgs that is fine, we can take that. Lets merge that first, and I'll rebase my code onto it.

**Review by ouphi — APPROVED** [review]

**Review by cpeeyush — APPROVED** [review]
Thanks @paullatzelsperger for using maven artifacts here it is a good progress and a milestone/achievement to use EDC from maven public repo. :) 

Approving this PR considering we will take fix https://github.com/agera-edc/IdentityHub/pull/45 for the broken test.

**Comment by paullatzelsperger** [comment]
@marcgs @cpeeyush due to a [pending PR](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1792) in EDC upstream it is better to indeed merge this variant of the fix, as it will accommodate the aforementioned upstream PR as well.
